### ISSUE TITLE
Addition of DenseNets to Keras applications

### ISSUE THREAD

**Issue by titu1994** [fchollet/keras#7845]
Recently, the weights for DenseNet have been ported to Keras from the Caffe version in https://github.com/flyyufelix/DenseNet-Keras. However, it depends on a custom Scale layer for BatchNorm, which is not needed in Keras.

Therefore I have transferred the weights to a model builder which does not require such custom Scale layers and there is currently a PR in Keras Contrib to update the DenseNet application there with the weights https://github.com/farizrahman4u/keras-contrib/pull/144.

@fchollet  Would it be better for Keras proper to have DenseNet as an application, and contrib to have just the DenseNet Fully Convolutional Network ? Or should both DenseNet and DenseNet-FCN remain in contrib for the time being?
